jax/numpy/lax_numpy.py

```diff
@@ -968,13 +968,21 @@ def ravel(a, order="C"):
 
 @_wraps(onp.squeeze)
 def squeeze(a, axis=None):
+  msg = "cannot select an axis to squeeze out " \
+            "which has size not equal to one"
   if 1 not in shape(a):
     return a
   if axis is None:
     newshape = [d for d in shape(a) if d != 1]
   else:
     if isinstance(axis, int):
+      if shape(a)[axis] != 1:
```

Comment:
Couldn't you avoid duplicating this check by simply waiting until after the `axis = (axis,)` call below and then making the test unconditional?

This would also allow you to move `msg` closer to its use.

Reply:
How can I check int and tuple same time?

Reply:
I mean you should write something like this:

```
axis = (axis,) if isinstance(axis, int) else axis
if any(shape(a)[x] != 1 for x in axis):
  raise ValueError("...")
```

Reply:
Okey! We are already converting to tuple if integer so two different queries are meaningless. I'm sorry to miss this. Thank you!